ArcGISRuntimeSDKQt_CppSamples/Routing/NavigateARouteWithRerouting/NavigateARouteWithRerouting.cpp

```diff
@@ -0,0 +1,369 @@
+
+QString defaultDataPath()
+{
+  QString dataPath;
+
+#ifdef Q_OS_ANDROID
+  dataPath = "/sdcard";
+#elif defined Q_OS_IOS
+  dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+#else
+  dataPath = QDir::homePath();
+#endif
+
+  return dataPath;
+}
+}
+
+NavigateARouteWithRerouting::NavigateARouteWithRerouting(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(BasemapStyle::ArcGISNavigation, this))
+{
+  m_routeOverlay = new GraphicsOverlay(this);
+
+  //setup route task
+  const QString folderLocation = QString("%1/ArcGIS/Runtime/Data/tpkx/san_diego").arg(defaultDataPath());
```

Comment:
it's a bit confusing that the geodatabase file is in the tpkx folder

Reply:
This is also the case for the Offline Routing sample, so instead of having the sample viewer download the files again to a new location I opted to reference where it would otherwise be placed.

Reply:
Ok. Not a huge issue, but we should log an issue to clean it up as it is misleading